src/utils/resin.ts

```diff
@@ -0,0 +1,28 @@
+const RESIN_PREFIX = 'data-resin-';
+
+export function getResinAttribute(name: string): string {
+    return `${RESIN_PREFIX}${name}`;
+}
+
+export function stringify(value: unknown): string | null {
+    if (typeof value === 'object' || typeof value === 'function') {
+        return null;
+    }
+
+    return typeof value === 'string' ? value : String(value);
```

Comment:
Should we use `toString` rather than the static constructor?

Reply:
I think `String` here is a global method not a constructor. And it's actually better than `toString` because it first calls `toString`, if fails, it will call `valueOf`.